engine/src/execution_graph/logic_controllers/BatchProcessing.h

```diff
@@ -163,11 +164,20 @@ class DataSourceSequence {
 		
 		batch_id++;
 		if (batch_id == all_row_groups[file_index].size()) {
+			is_csv = false;
```

Comment:
is this the only state possible here csv true or false ? arent there many different things it could be like orc or parquet

Reply:
when come from a ORC o PARQUET file satisfy that condition and then is_csv set to false. In other case (csv) just jump it.